consensus/core/src/dag_state.rs

```diff
@@ -27,48 +28,65 @@ const BLOCK_CACHED_ROUNDS: Round = 100;
 pub(crate) struct DagState {
     context: Arc<Context>,
 
-    // Caches uncommitted blocks, and recent blocks within BLOCK_CACHED_ROUNDS from the
-    // last committed leader round.
+    // Caches blocks within CACHED_ROUNDS from the last committed round per authority.
+    // Note: uncommitted blocks will always be in memory.
     recent_blocks: BTreeMap<BlockRef, VerifiedBlock>,
 
-    // All accepted blocks have their refs cached. Cached refs are never removed for now.
-    // Each element in the vector contains refs for the authority corresponding to its index.
+    // Accepted blocks have their refs cached. Cached refs are never removed until restart.
+    // Each element in the Vec corresponds to the authority with the index.
     cached_refs: Vec<BTreeSet<BlockRef>>,
 
+    // Last consensus commit of the dag.
+    last_commit: Option<Commit>,
+
     // Persistent storage for blocks, commits and other consensus data.
     store: Arc<dyn Store>,
 }
 
 #[allow(unused)]
 impl DagState {
-    pub(crate) fn new(
-        context: Arc<Context>,
-        blocks: Vec<VerifiedBlock>,
-        store: Arc<dyn Store>,
-    ) -> Self {
+    /// Initializes DagState from storage.
+    pub(crate) fn new(context: Arc<Context>, store: Arc<dyn Store>) -> Self {
```

Comment:
I assume you'll that on follow up PRs, but leaving as comment anyways for test the recovery/inti logic